### ISSUE TITLE
Basalt in Expert mode

### ISSUE THREAD

**Issue by henkekalmar** [stanhebben/MineTweaker3#423]
Hello, I've been trying to add a basalt recipe onto my modded Infinity evolved server. And I grab the basalt I want from ingame(the cobblestone one), put it in my hand and do /mt hand and write that exact name into my file. Add it but I still get an error. (error: https://gyazo.com/f1ba593cc7a66419f729d3816afbace0 ) 

`#Name: Henk.zs
#Author: Henk

print("Initializing 'Henk.zs'...");

recipes.addShaped(<ProjRed:Exploration:projectred.exploration.stone:2> * 4, [[<Railcraft:cube:4>, <minecraft:stone>, <Railcraft:cube:4>], [<minecraft:stone>, <ThermalFoundation:material:1029>, <minecraft:stone>], [<Railcraft:cube:4>, <minecraft:stone>, <Railcraft:cube:4>]]);

print("Initialized 'Henk.zs'");`
Replacing the basalt with anything else seem to work just fine. For some reason it's not liking it and I can't figure out why... Please help! :-)

**Comment by jaredlll08** [comment]
Project red breaks the modid naming scheme by having an extra ":" in it,
nothing we can do

**Comment by henkekalmar** [comment]
So it's impossible for me to add a recipe resulting in, or including any project red item?

**Comment by jaredlll08** [comment]
Correct, once again, this is not an issue on our side, this is an issue
with how project red has named their modid, there is nothing we can do to
fix it, and project red also will not fix it as changing their modid will
remove blocks from the game (blocks are stored in the world based on
modid:name, so changing that with result in the old blocks being replaced)

**Comment by henkekalmar** [comment]
Alright. Cheers, thanks for your help.

**Comment by Blue64** [comment]
i think there might perhaps be a way to handle it, but it's similar to the Fluid Names with spaces, kind of a pain to jump through all of those hoops just to concatenate the `:` symbol so you can bypass the check.

there might also be a way to do it if you count how many `:` are found and ~use the last/second one found~ do a try loop on each `:` as the divider, actually, thinking about it, you'd pretty much need to concatenate the ~first one~ remaining ones out anyways, but it would be easier that way as you'd be able to simply run an If/Else to locate a valid substitute, perhaps utilize an error catcher at the end to say something like
```
if error != null && %s1 != null (
  if %s1 contains "." (
    try for every "." in %s1 (
      StringReplace %s1 to %s2 replace "." with ":"
      attempt to see if %s2 has a valid Item output
    )
  )
)
```
my Java is super rusty, but I'm fairly sure you can understand what I'm getting at there., you might need to process the  ModID separate from the ItemName/BlockName, preserving the location of the valid divider.

I know there's some possible way to check. perhaps a conversion back to `:` after the Compiler/Tokenator?

Did some Edits to correct typos & whatnot.

**Comment by Blue64** [comment]
did some edits (for those reading via email)

**Comment by Blue64** [comment]
also, @henkekalmar you can try reading my GitHub Formatting Tips:
`>>` [Here](https://github.com/Blue64/Misc/issues/2) `<<`
(I sometimes forget it, so I made a post in a misc Repo about it lol)
